pilot/test/server/echo/handler.go

```diff
@@ -0,0 +1,227 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package echo
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/gorilla/websocket"
+	"google.golang.org/grpc/metadata"
+
+	pb "istio.io/istio/pilot/test/grpcecho"
+	"istio.io/istio/pkg/log"
+)
+
+type handler struct {
+	port    int
+	version string
+}
+
+var upgrader = websocket.Upgrader{
+	CheckOrigin: func(r *http.Request) bool {
+		// allow all connections by default
+		return true
+	},
+} //defaults
+
+// Imagine a pie of different flavors.
```

Comment:
interesting analogy

Reply:
Haha .... not my comment ... I've never seen this before.  But hilarious! :)